src/coreclr/src/jit/optcse.cpp

```diff
@@ -402,10 +402,27 @@ void Compiler::optValnumCSE_Init()
 //
 unsigned Compiler::optValnumCSE_Index(GenTree* tree, Statement* stmt)
 {
-    unsigned key;
+    ssize_t  key;
     unsigned hash;
     unsigned hval;
     CSEdsc*  hashDsc;
+    bool     isIntConstHash       = false;
+    bool     enableSharedConstCSE = false;
+    int      configValue          = JitConfig.JitDisableConstCSE();
+
+#if defined(TARGET_ARM64)
+    // ARM64 - allow to combine with nearby offsets, when config is not 2
+    if (configValue != 2)
+    {
+        enableSharedConstCSE = true;
+    }
+#endif // TARGET_ARM64
+
+    // All Platforms - also allow to combine with nearby offsets, when config is 3
+    if (configValue == 3)
```

Comment:
>configValue [](start = 8, length = 11)

are you considering adding a JITStress pipeline or something equivalent to test this config value?

Reply:
@TamarChristinaArm  Yes CORINFO_HELP_ARRADDR_ST has no return value, but it follows the normal calling convention, which kills the incoming argument values in x0, w1 and x2.  So they have to be reloaded at each call site.

Reply:
I wasn't planning on adding that.  I  want to be able to experiment on enabling this for the other platforms.